### ISSUE TITLE
Map for Sileno Smart movers

### ISSUE THREAD

**Issue by kizio1000** [py-smart-gardena/hass-gardena-smart-system#177]
Hi,

For some time now, there are smart movers from Sileno Life Smart LONA line that provide GPS map of mover working (like in Husquarna products). Map can be viewed in Garden app. Is it possible to add view of this map to gardena mover integration ? 

![SmartSelect_20230526_104150_GARDENA smart system](https://github.com/py-smart-gardena/hass-gardena-smart-system/assets/76654878/09037297-914f-4440-b21b-30aa4d5348f8)

**Comment by Jogge** [comment]
I agree that this feature has the potential to enhance the integration.

Unfortunately, as of now, the [GARDENA smart system API](https://developer.husqvarnagroup.cloud/apis/gardena-smart-system-api?tab=openapi) does not include GPS position data. However, the [Automower Connect API](https://developer.husqvarnagroup.cloud/apis/automower-connect-api?tab=openapi) already provides access to GPS position data. We can only hope that Husqvarna also considers adding GPS position data in a future API update for the [GARDENA smart system API](https://developer.husqvarnagroup.cloud/apis/gardena-smart-system-api?tab=openapi).

In the meantime, it would be helpful to communicate your interest in this feature to Husqvarna. You can provide feedback and express your longing for GPS position data in a future API release by reaching out to Husqvarna via email at [feedback@developer.husqvarnagroup.cloud](mailto:feedback@developer.husqvarnagroup.cloud). Your input can play a vital role in shaping the direction of their development efforts.

**Comment by kizio1000** [comment]
I did communicated with them, they seamed interested but for quite some time no change. Too bad as this APT change would not influence competition among husqarna and agrdena brands as it's purely nerdish stuff.

![image](https://github.com/py-smart-gardena/hass-gardena-smart-system/assets/76654878/4ac2718c-0932-428c-aa6d-4e2b16d84083)

**Comment by tbnobody** [comment]
The API for the web api (e.g. https://smart.gardena.com/v1/garden-map/postprocessing/mowers/xxxxxxxxxxxx/map) already exposes several information. I am wondering why they are exposing the information in the official API as well

**Comment by Fywqdk** [comment]
As far as I can see there is several location-related endpoints in the v2 API as well. I am not particularly well versed in stuff like this but I can try to find a couple of hours to poke around and see what information I can get.

**Comment by sthomsen1** [comment]
From what I’ve seen so far, the location-related endpoints don’t seem to refer to actual GPS coordinates of the devices, but rather to logical areas – like defining where a mower or smart plug is located (e.g. “my garden”).
I was hoping to get access to GPS data or, at the very least, the zones that are shown in the app – but unfortunately, I haven’t found anything like that exposed in the API so far.

**Comment by Fywqdk** [comment]
Good to know. Honestly haven't had time to look at it yet. I did try to suggest it to Husqvarna/Gardena a while back but they didn't even respond to me :/

**Comment by andy778** [comment]
I had idea to get out GPS signals from the SSH and setup local API e.g MQTT and get out data 
https://github.com/husqvarnagroup/smart-garden-gateway-public

and then maps or heat maps 
- radio signal, ( If I need to move the 868MHZ transmitter alias gateway) 
- Where mower have travelled, then one could mark out obstacles 
- If the Z cordinate of GPS is good enough then where I need to go and fill up with sand, or then just see where the mower have changed direction always

**Comment by sthomsen1** [comment]
I have some new findings, that might be useful for this integration, even if we don't have full GPS coordinates yet.

I’ve been looking for a way to send my Gardena mower to specific zones. While I don’t have a graphical map in my local setup yet, I managed to reverse-engineer the commands used by the Gardena Web-App (smart.gardena.com).

By monitoring the network traffic while interacting with the web interface, I noticed that it still uses the v1 backend. When manually starting the mower for a specific area, the app sends a POST request to the following endpoint:

https://smart.gardena.com/v1/devices/YOUR-DEVICE-ID/abilities/mower/commands/manual_start?locationId=YOUR-LOCATION-ID

Example Payload (json):
```
{
  "mowerTimer": 7200,
  "startingPointDistance": 1,
  "areaId": null
}
```

startingPointDistance seems to be the crucial value. By triggering a start for each of my defined zones in the web app and checking the network logs, I was able to identify the specific "distance" value associated with each zone (e.g., "Front Garden" vs. "Back Garden").
Using these values, I can now programmatically send the mower to different zones with a simple request. It’s currently an experimental setup, but it proves that zone-based steering is possible via the existing web API infrastructure.

Regarding the Mapping: The web app also seems to fetch data to render the map dynamically. While I haven't fully parsed the map generation yet, the traffic suggests that the coordinates or pathing data is indeed being transmitted to the client, which we might be able to intercept as well.

This could be a viable workaround for zone control until (or if) Husqvarna officially exposes these LONA features in the public developer API.
